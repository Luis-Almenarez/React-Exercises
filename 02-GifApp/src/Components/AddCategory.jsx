import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    const textUser = inputValue.trim();
    if (textUser.length <= 2) return;
    onNewCategory(textUser.toLocaleLowerCase());
    // setCategories((categories) => [inputValue, ...categories]);
    setInputValue("");
  };
  return (
    <>
      <form onSubmit={onSubmit}>
        <input
          type="text"
          placeholder="Search Gif"
          value={inputValue}
          onChange={onInputChange}
        />
      </form>
    </>
  );
};
