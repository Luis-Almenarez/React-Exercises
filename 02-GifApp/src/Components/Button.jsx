export const Button = ({ text, onClick }) => {
  return (
    <button
      onClick={onClick}
      className="bg-indigo-500 hover:bg-indigo-700 transition-all text-white text-sm font-bold py-1 px-6 rounded focus:outline-none uppercase">
      {text}
    </button>
  );
};
