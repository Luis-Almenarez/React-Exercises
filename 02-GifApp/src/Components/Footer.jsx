export const Footer = () => {
  return (
    <footer className="w-full flex pt-10 gap-1 justify-center font-mono italic">
      <p className="text-sm font-bold">Created for Â©</p>
      <a
        className="text-sm font-bold transition-all hover:text-indigo-600"
        href="https://github.com/luisalmenarez"
        target="_blank"
        rel="noreferrer">
        Luis Almenarez
      </a>
      <p className="text-sm font-bold italic">2023</p>
    </footer>
  );
};
