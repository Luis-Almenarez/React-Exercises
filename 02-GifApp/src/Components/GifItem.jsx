import PropTypes, { string } from "prop-types"; // Importamos la dependecia prop-types

export const GifItem = ({ title, url, id }) => {
  return (
    <div className="grid place-items-center bg-white text-black font-bold rounded-lg overflow-hidden ">
      <img className="w-full h-full" src={url} alt={title} />
      <p className="bg-white text-sm m-1 font-bold p-1">{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
