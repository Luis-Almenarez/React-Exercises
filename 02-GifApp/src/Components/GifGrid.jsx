import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3 className="font-bold pt-8 pb-8 p-0">{category}</h3>
      {isLoading && <h2 className="text-4xl font-bold m-5">Searching...</h2>}
      <section className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6 place-items-center w-full">
        {images.map((image) => (
          <div
            key={image.id}
            className="grid w-full h-56 transition-all hover:scale-105">
            <GifItem {...image} />
          </div>
        ))}
      </section>
    </>
  );
};
