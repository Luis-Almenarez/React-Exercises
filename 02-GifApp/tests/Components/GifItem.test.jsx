import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/Components/GifItem";

describe("Pruebas en GifItem.test.jsx", () => {
  const title = "Naruto";
  const url = "https://naruto-shippuden/Naruto.jpg";

  test("Debe hacer match con el Snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el URL y el atributo ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole("img").src).toBe(url);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Dbe mostrar el tÃ­tulo en el componente", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
