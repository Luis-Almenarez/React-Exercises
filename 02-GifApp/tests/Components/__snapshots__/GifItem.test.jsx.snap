// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem.test.jsx Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="grid place-items-center bg-white text-black font-bold rounded-lg overflow-hidden "
  >
    <img
      alt="Naruto"
      class="w-full h-full"
      src="https://naruto-shippuden/Naruto.jpg"
    />
    <p
      class="bg-white text-sm m-1 font-bold p-1"
    >
      Naruto
    </p>
  </div>
</div>
`;
